builtin/providers/kubernetes/structures_pod.go

```diff
@@ -0,0 +1,494 @@
+	}
+	in := l[0].(map[string]interface{})
+	obj := &v1.SecretVolumeSource{
+		SecretName: in["secret_name"].(string),
+	}
+	return obj
+}
+
+func expandVolumes(volumes []interface{}) ([]v1.Volume, error) {
+	if len(volumes) == 0 {
+		return []v1.Volume{}, nil
+	}
+	vl := make([]v1.Volume, len(volumes))
+	for i, c := range volumes {
+		m := c.(map[string]interface{})
+
+		if value, ok := m["name"]; ok {
+			vl[i].Name = value.(string)
+		}
+		if value, ok := m["persistent_volume_claim"].([]interface{}); ok && len(value) > 0 {
+			vl[i].PersistentVolumeClaim = expandPersistentVolumeClaimVolumeSource(value)
+		}
+		if value, ok := m["secret"].([]interface{}); ok && len(value) > 0 {
+			vl[i].Secret = expandSecretVolumeSource(value)
+		}
+		if v, ok := m["gce_persistent_disk"].([]interface{}); ok && len(v) > 0 {
+			vl[i].GCEPersistentDisk = expandGCEPersistentDiskVolumeSource(v)
+		}
+		if v, ok := m["aws_elastic_block_store"].([]interface{}); ok && len(v) > 0 {
+			vl[i].AWSElasticBlockStore = expandAWSElasticBlockStoreVolumeSource(v)
+		}
+		if v, ok := m["host_path"].([]interface{}); ok && len(v) > 0 {
+			vl[i].HostPath = expandHostPathVolumeSource(v)
+		}
+		if v, ok := m["glusterfs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Glusterfs = expandGlusterfsVolumeSource(v)
+		}
+		if v, ok := m["nfs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].NFS = expandNFSVolumeSource(v)
+		}
+		if v, ok := m["rbd"].([]interface{}); ok && len(v) > 0 {
+			vl[i].RBD = expandRBDVolumeSource(v)
+		}
+		if v, ok := m["iscsi"].([]interface{}); ok && len(v) > 0 {
+			vl[i].ISCSI = expandISCSIVolumeSource(v)
+		}
+		if v, ok := m["cinder"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Cinder = expandCinderVolumeSource(v)
+		}
+		if v, ok := m["ceph_fs"].([]interface{}); ok && len(v) > 0 {
+			vl[i].CephFS = expandCephFSVolumeSource(v)
+		}
+		if v, ok := m["fc"].([]interface{}); ok && len(v) > 0 {
+			vl[i].FC = expandFCVolumeSource(v)
+		}
+		if v, ok := m["flocker"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Flocker = expandFlockerVolumeSource(v)
+		}
+		if v, ok := m["flex_volume"].([]interface{}); ok && len(v) > 0 {
+			vl[i].FlexVolume = expandFlexVolumeSource(v)
+		}
+		if v, ok := m["azure_file"].([]interface{}); ok && len(v) > 0 {
+			vl[i].AzureFile = expandAzureFileVolumeSource(v)
+		}
+		if v, ok := m["vsphere_volume"].([]interface{}); ok && len(v) > 0 {
+			vl[i].VsphereVolume = expandVsphereVirtualDiskVolumeSource(v)
+		}
+		if v, ok := m["quobyte"].([]interface{}); ok && len(v) > 0 {
+			vl[i].Quobyte = expandQuobyteVolumeSource(v)
+		}
+		if v, ok := m["azure_disk"].([]interface{}); ok && len(v) > 0 {
+			vl[i].AzureDisk = expandAzureDiskVolumeSource(v)
+		}
+		if v, ok := m["photon_persistent_disk"].([]interface{}); ok && len(v) > 0 {
+			vl[i].PhotonPersistentDisk = expandPhotonPersistentDiskVolumeSource(v)
+		}
+	}
+	return vl, nil
+}
+
+func expandPodTemplateSpec(p []interface{}) (v1.PodTemplateSpec, error) {
```

Comment:
As far as I can see this function is currently unused? 👀  I'm happy for you to help with the implementation of RC 😃 , but I'd prefer if kept this PR only focused on Pod.